text/0000-buildpack-author-subteam.md

```diff
@@ -0,0 +1,125 @@
+
+## Sub-team Guidelines
+
+The Buildpack Author sub-team will follow the same guidelines as other sub-teams:
+
+- include at least one core team member
+- responsible for narrower sets of concerns related to specific aspects of the project
+
+## Language Binding Responsibilties
+These are the responsibilities for maintaining any language binding:
+1. Implementing new APIs in a timely manner
+1. Release process, planning and updates
+1. Triaging issues and supporting users
+1. Documentation
+1. Buildpack Examples using these bindings
+1. Planning and executing improvements
+
+Today, this means taking ownership over the `libcnb` repo doing the above for it.
+
+## Inline Buildpack Responsibilites
+The buildpack sub-team is responsible for sheparding and maintaining [Inline Buildpacks](https://github.com/buildpacks/rfcs/blob/main/text/0048-inline-buildpack.md).
+
+## Collaboration with other teams
+The buildpack sub-team should engage with the learning team in order to:
+1. Improve documentation for buildpack authors
+1. Document buildpack best practices
+
+The buildpack sub-team should provide feedback on RFCs that change the buildpack API with an eye to the potential implications for buildpack authors.
+
+## Future Scope
+The following are examples of additional tools that would fall to the purview of the buildpack team:
+1. Language bindings for other languages
+1. Project Templates for bootstrapping buildpacks
+1. A test harness for integration testing buildpacks
+1. A DSL for writing buildpacks (e.g. [packfile](https://github.com/sclevine/packfile) or similar)
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+- Yet another sub-team
+- May be difficult to sustain maintainers for all these teams
+- May create a silo instead of spreading out the buildpack author concerns
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Do Nothing
+
+If this team isn't made, `libcnb` will continue to live in the implementation team. That sub-team can then staff more focused contributors/maintainers for maintaing that library. Each other "Buildpack Author" focused project/tool will need to find an appropriate home. This will fragment and hurt the focus of having a single team built around it.
+
+## Distribution Team
+
+The distribution team already handles a part of the Buildpack Author experience which is publishing a buildpack and is currently a fairly small team. This would let the two sub-teams combine with expanded scope. This isn't a perfect match though, since the distribution team is well scoped at this point focusing on the registry, interacting with it as users, and operating it.
+
+# Prior Art
+[prior-art]: #prior-art
+
+* [Rust Dev tools team](https://www.rust-lang.org/governance/teams/dev-tools)
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- Who are the maintainers?
+- Would this sub-team own the pack buildpack author tooling (like `pack buildpack new` and inline buildpacks)?
```

Comment:
I think that partitioning aspects of `pack` is likely to cause things to slip between the crack. If anything, making `pack buildpack new` a very slim interface to some `libcnb` function that is called would be fantastic.

Reply:
yeah, I think this team probably more than others may be doing cross team work whether in `pack` or `lifecycle` like you mentioned. I'm not entirely sure how we're going to thread that needle, but one way for `pack buildpack new` would be this sub-team would own the template repo and just PR stuff into `pack`, but ultimately platform would own the CLI side.

Reply:
This was another aspect we were thinking about. I was hoping pack could simply house a templating library like https://github.com/tmrts/boilr or https://github.com/cookiecutter/cookiecutter that could just reference the latest version of the official templates.

That way the templating logic in pack could be generic and the BP author team could be responsible for the upkeep of the templates. This could also allow pack to allow using community maintained templates.

Just a thought - - nothing concrete but I imagine this could provide a nice separation of concerns :)

cc @jromero and @jkutner (I believe you created the original rfc around this)

Reply:
I agree with @dfreilich that responsibilities should be assigned at the component level. But as @samj1912 mentioned would could certainly extract buildpack author focused tool from pack (libraries or resources like templates) and give this new team responsibility for them.

Reply:
I tried to addressed this in d979f93.